internal/hncconfig/reconciler.go

```diff
@@ -200,7 +204,7 @@ func (r *Reconciler) reconcileConfigTypes(inst *api.HNCConfiguration, allRes []*
 			log := r.Log.WithValues("resource", gr, "appliedMode", gvkMode.mode)
 			msg := ""
 			// Set a different message if the type is enforced by HNC.
-			if api.IsEnforcedType(rsc) {
+			if !config.EnforcedTypesDisabled && api.IsEnforcedType(rsc) {
```

Comment:
Ditto, we should do this check inside `api.IsEnforcedType`. That may mean moving the global var from `config` to `api` instead since I think `config` imports `api` but not vice versa.